test/api/detection_test.cpp

```diff
@@ -0,0 +1,94 @@
+#include <fstream>
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/cigar/cigar.hpp>
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/io/exception.hpp>
+#include <seqan3/io/sam_file/sam_flag.hpp>
+
+#include "iGenVar.hpp"
+
+#include "modules/sv_detection_methods/analyze_cigar_method.hpp"    // for the split read method
+#include "modules/sv_detection_methods/analyze_sa_tag_method.hpp"   // for the cigar string method
+#include "structures/junction.hpp"                                  // for class Junction
+
+using seqan3::operator""_cigar_operation;
+using seqan3::operator""_dna5;
+
+/* -------- detection methods tests -------- */
+
+// TODO (irallia): implement test cases
+
+TEST(junction_detection, cigar_string_method_simple)
+{
+    std::string const read_name = "read021";
+    std::string const chromosome = "chr1";
+    int32_t const query_start_pos = 1;
+    std::vector<seqan3::cigar> cigar_sting = {{5, 'S'_cigar_operation}, {15, 'M'_cigar_operation},
+                                              {6, 'D'_cigar_operation}, {24, 'M'_cigar_operation}}; //5S15M6D24M
+    seqan3::dna5_vector seq = {"GGGCTCATCGATCGATTTCGGATCGGGGGGCCCCCATTTTAAACGGCCCC"_dna5};
+    std::vector<Junction> junctions_res{};
+    uint64_t const min_var_length = 5;
+    analyze_cigar(read_name, chromosome, query_start_pos, cigar_sting, seq, junctions_res, min_var_length);
+
+    Breakend new_breakend_1 {chromosome, 16, strand::forward};
+    Breakend new_breakend_2 {chromosome, 22, strand::forward};
+    std::vector<Junction> junctions_expected_res{Junction{new_breakend_1, new_breakend_2, ""_dna5, read_name}};
+
+    EXPECT_EQ(junctions_expected_res.size(), junctions_res.size());
+
+    for (size_t i = 0; i < junctions_expected_res.size(); ++i)
+    {
+        EXPECT_EQ(junctions_expected_res[i].get_read_name(), junctions_res[i].get_read_name());
+        EXPECT_TRUE(junctions_expected_res[i] == junctions_res[i]);
+    }
+}
+
+TEST(junction_detection, split_read_method_simple)
+{
+    std::string const read_name = "read021";
+    seqan3::sam_flag const flag{0u};
+    std::string const chromosome = "chr1";
+    int32_t const pos = 80;
+    uint8_t const mapq = 60;
+    std::vector<seqan3::cigar> cigar_sting = {{46, 'M'_cigar_operation}, {4, 'S'_cigar_operation}}; //46M4S
+    seqan3::dna5_vector seq = {"GGGCTCATCGATCGATTTCGGATCGGGGGGCCCCCATTTTAAACGGCCCC"_dna5};
+    std::string const sa_tag = "SA:Z:chr1,108,+,44S6M,60,6,0;";
```

Comment:
The two CIGAR strings indicate a slight 2bp overlap on the read (i.e. the 45. and 46. read base are aligned twice). It is checked [here](https://github.com/seqan/iGenVar/blob/f92d9c24de496d35a24721b9b031ebb50323f601/src/modules/sv_detection_methods/analyze_sa_tag_method.cpp#L65) that no such overlap exists before detecting a junction. So if you remove the overlap, `junction_res` should contain a junction which you can test for in `junctions_expected_res`.

Reply:
Thanks for the hint. I found another mistake in our mini_example: https://github.com/seqan/iGenVar/issues/108